src/components/ElectedRepresentative/CreateEditModal.jsx

```diff
@@ -0,0 +1,300 @@
+import { useNavigate } from 'react-router'
+
+const Title = styled(Typography)(
+  ({ theme }) => `
+  color: ${theme.palette.colors.gray[900]};
+  font-size: 1.525rem;
+  line-height: 1.75rem;
+  font-weight: 500;
+`
+)
+
+const FormTitle = styled(Typography)(
+  ({ theme }) => `
+  display: block;
+  color: ${theme.palette.colors.gray[900]};
+  font-size: 0.975rem;
+  line-height: 1.5rem;
+  font-weight: 500;
+`
+)
+
+const FormDescription = styled(Typography)(
+  ({ theme }) => `
+  display: block;
+  color: ${theme.palette.colors.gray[500]};
+  font-size: 0.875rem;
+  line-height: 1.25rem;
+  margin-top: 0.25rem;
+`
+)
+
+export const FormHeading = ({ title, description }) => (
+  <Box sx={{ pb: 1.5, mb: 2, borderBottom: '1px solid', borderBottomColor: theme => theme.palette.colors.gray[300] }}>
+    <FormTitle>{title}</FormTitle>
+    <FormDescription>{description}</FormDescription>
+  </Box>
+)
+
+FormHeading.propTypes = {
+  title: PropTypes.string,
+  description: PropTypes.string,
+}
+
+const messages = {
+  create: 'Créer',
+  update: 'Modifier',
+  close: 'Fermer',
+  creationTitle: "Creation d'un élu",
+  editionTitle: "Modification de l'élu",
+  createSuccess: 'Elu créée avec succès',
+  editSuccess: "L'élu a bien été modifiée",
+}
+
+const fields = {
+  firstName: 'first_name',
+  lastName: 'last_name',
+  gender: 'gender',
+  birthDate: 'birth_date',
+  birthPlace: 'birth_place',
+  contactPhone: 'contact_phone',
+  contactEmail: 'contact_email',
+  hasFollowedTraining: 'has_followed_training',
+}
+
+const electedSchema = Yup.object({
+  firstName: Yup.string().required('Le Nom est obligatoire'),
+  lastName: Yup.string().required('Le Prenom est obligatoire'),
+  gender: Yup.string().required('Le genre est obligatoire'),
+  birthDate: Yup.date().required('La date de naissance est obligatoire'),
+  contactEmail: Yup.string().email("L'email est invalide"),
+})
+
+const CreateEditModal = ({ elected, handleClose, onCreateResolve, onUpdateResolve }) => {
+  const [selectedAdherent, setSelectedAdherent] = useState(null)
+  const { enqueueSnackbar } = useCustomSnackbar()
+  const navigate = useNavigate()
+  const { handleError, errorMessages } = useErrorHandler()
+  const { control, getValues, reset, watch } = useForm({
+    mode: 'onChange',
+    resolver: yupResolver(electedSchema),
+  })
+
+  watch()
+
+  const { mutate: createOrUpdate, isLoading } = useMutation(!elected ? createElected : updateElected, {
+    onSuccess: elected => {
+      onCreateResolve && onCreateResolve()
+      onUpdateResolve && onUpdateResolve()
+      enqueueSnackbar(elected ? messages.createSuccess : messages.editSuccess, notifyVariants.success)
+      handleClose()
+      navigate(`/elected_representatives/${elected.uuid}`)
+    },
+    onError: handleError,
+  })
+
+  const values = getValues()
+
+  const createOrEdit = () => {
+    createOrUpdate({ ...values, adherent: selectedAdherent.uuid, uuid: elected?.uuid })
```

Comment:
est-ce qu'on est sûr de toujours avoir un adherent sélectionné? 
sinon:  

```suggestion
    createOrUpdate({ ...values, adherent: selectedAdherent?.uuid, uuid: elected?.uuid })
```